schedulers/argo-workflow/addons.tf

```diff
@@ -0,0 +1,106 @@
+  # Metrics Server
+  #---------------------------------------------------------------
+  enable_metrics_server = true
+
+
+  #---------------------------------------------------------------
+  # Cluster Autoscaler
+  #---------------------------------------------------------------
+  enable_cluster_autoscaler = true
+
+
+  #---------------------------------------------------------------
+  # Spark Operator Add-on
+  #---------------------------------------------------------------
+  enable_spark_k8s_operator = true
+
+  #---------------------------------------------------------------
+  # Apache YuniKorn Add-on
+  #---------------------------------------------------------------
+  enable_yunikorn = true
+
+
+  depends_on = [
+    module.eks_blueprints
+  ]
+  tags = local.tags
+
+}
+
+
+#-------------------------------------------------
+# Argo Workflows Helm Add-on
+#-------------------------------------------------
+locals {
+
+  default_helm_config = {
+    name             = "argo-workflows"
+    chart            = "argo-workflows"
+    repository       = "https://argoproj.github.io/argo-helm"
+    version          = "v0.20.1"
+    namespace        = "argo-workflows"
+    create_namespace = true
+    description      = "Argo workflows Helm chart deployment configuration"
+  }
+
+
+
+  irsa_config = {
+    kubernetes_namespace              = local.default_helm_config["namespace"]
+    kubernetes_service_account        = local.name
+    create_kubernetes_namespace       = try(local.default_helm_config["create_namespace"], true)
+    create_kubernetes_service_account = false
+    irsa_iam_policies                 = []
+  }
+
+  eks_oidc_issuer_url = replace(data.aws_eks_cluster.eks_cluster.identity[0].oidc[0].issuer, "https://", "")
+}
+
+module "helm_addon" {
+  source      = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons/helm-addon?ref=v4.12.0"
+  helm_config = local.default_helm_config
+  irsa_config = local.irsa_config
+
+  addon_context = {
+    aws_caller_identity_account_id = data.aws_caller_identity.current.account_id
+    aws_caller_identity_arn        = data.aws_caller_identity.current.arn
+    aws_eks_cluster_endpoint       = module.eks_blueprints.eks_cluster_endpoint
+    aws_partition_id               = data.aws_partition.current.partition
+    aws_region_name                = data.aws_region.current.name
+    eks_cluster_id                 = module.eks_blueprints.eks_cluster_id
+    eks_oidc_issuer_url            = replace(data.aws_eks_cluster.eks_cluster.identity[0].oidc[0].issuer, "https://", "")
+    eks_oidc_provider_arn          = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${local.eks_oidc_issuer_url}"
+    tags                           = local.tags
+    irsa_iam_role_path             = "/"
+    irsa_iam_permissions_boundary  = ""
+  }
+
+  depends_on = [
+    module.eks_blueprints
+  ]
+
+}
```

Comment:
Move this block to `addons.tf` file

Reply:
I think it would be better if you add this Argo Workflow Helm Chart add-on to EKS Blueprints and refer here